src/Services/TupleFilterService.php

```
@@ -106,14 +104,28 @@
-        return sprintf(
-            '%s#%s@%s%s',
-            $tuple->getUser() ?? '',
-            $tuple->getRelation() ?? '',
-            $tuple->getObject() ?? '',
-            $condition instanceof ConditionInterface ? '#' . spl_object_hash($condition) : '',
-        );
+        if ($condition instanceof ConditionInterface) {
+            $keyData['cond_name'] = $condition->getName();
+            $context = $condition->getContext();
+
+            if (! empty($context)) {
```

PHPStan (2.1.15)

Construct empty() is not allowed. Use more strict comparison.


(empty.notAllowed)